python/unityagents/curriculum.py

```diff
@@ -29,14 +31,25 @@ def __init__(self, location, default_reset_parameters):
             except UnicodeDecodeError:
                 raise UnityEnvironmentException("There was an error decoding {}".format(location))
             self.smoothing_value = 0
-            for key in ['parameters', 'measure', 'thresholds',
-                        'min_lesson_length', 'signal_smoothing']:
+            necessaryParams = ['parameters', 'measure',
+                        'min_lesson_length', 'signal_smoothing']
+            if(self.data['measure'] == "multibrain_reward"): 
+                self.smoothing_values = {}
+                necessaryParams.append('threshold_operator')
+                necessaryParams.append('threshold_amount')
+            else:    
+                necessaryParams.append('thresholds')
+            for key in necessaryParams:
```

Comment:
You call `self.data['measure'] ` before checking if the key is in the dictionary

Reply:
Oh wow. 
Wasn't aware of this document being changed.
All the changes that I meant to make were in trainer_controller.py and curriculum only.